### ISSUE TITLE
Improvements to the online library (LOL)

### ISSUE THREAD

**Issue by Theverat** [LuxCoreRender/BlendLuxCore#522]
Related to #337

# GPU Memory Usage

* My OS uses about 0.2 GB of GPU RAM.
* With Blender open, 0.3 GB usage are reported (so Blender uses about 0.1 GB)
* After opening the asset bar and scrolling through all model and material asset once (so all thumbnails were shown), a usage of 1.7 GB is reported. So the asset bar thumbnails need about 1.4 GB of GPU memory, when all of them are loaded.

![Capture](https://user-images.githubusercontent.com/1584761/88454740-e3ff3100-ce71-11ea-94d5-6699d11501e6.PNG)

To say it frankly, this is unacceptable. 
In my [Image Node Preview](https://blendermarket.com/products/image-node-preview) addon, I can store 100 thumbnails of similar display size in 25.6 MiB.

The main problem seems to be that the thumbnails are 1080x1080 pixels in original size, although they are never shown that large. I would propose to create downscaled versions and only load those into Blender.

# Disabled code

There's a lot of disabled (commented) code - please enable and use, or delete. Thanks to git, deleted code is never lost.

# Properties

The file `properties/lol/LuxCoreOLScene.py` contains several PropertyGroups that are used by LOL. 
Many of the Properties in these groups should not be properties in the first place: Properties should only be used for variables that either need to be serialized and saved in the .blend file, or need to be exposed for Blender's operator API.

To me, it looks like neither is the case for many of the members in the LOL file. They should be regular Python member variables instead.

**Comment by Theverat** [comment]
@neo2068 Could you look into the issues listed here at some point? No pressure, doesn't need to happen immediately, just something to keep in mind during the further development of LOL.

**Comment by neo2068** [comment]
Yes, sure. The first implementation is based on the blenderkit addon and they did it that way. Now, we can change the implementation to our needs and think about some implementation details. Your feedback is very welcome as you know the blender python api and the BlendLuxCore addon the best. I will add all the requested features here to get track of it.

**Comment by neo2068** [comment]
How do you perform the thumbnail generation in your addon? I tried to downscale the preview images after loading but Blender wants to save the changed images when it gets closed.

**Comment by Theverat** [comment]
I would use some image batch processing to run the downscaling once, then store them in an extra folder. 
You can do the batch processing in something like Photoshop, or write a Python script to do it.

**Comment by neo2068** [comment]
The preview files are now downscaled to 128x128. The big preview in the infobox is only stored temporarily in memory and only one at once.

**Comment by Theverat** [comment]
Awesome. Can confirm that memory usage of the asset bar is now only a few megabytes even after scrolling through the whole list.

**Comment by CodeFHD** [comment]
Closing this issue after a review with the following reasons/conclusions:
- Issue could not be reproduced using latest version.
- No GPU memory changes scrolling through the asset bar and looking at large preview (tooltip)
- Comments mention changes around/before 20.08.2020. Possibly https://github.com/LuxCoreRender/BlendLuxCore/commit/ffe8e26490cea5c2b0077dc726e8f3436c2bc90e as the relevant commit
- Concluding that the issue was fixed

If you feel this issue is still relevant in 2025, please re-test it with a recent Blender + LuxCore version, then ask to reopen and provide the following information:
- Operating System
- CPU + GPU model
- Blender and LuxCoreRender version used for re-test
- Detailed steps how to reproduce the bug from an empty scene,
- provide a specific .blend-file
